### PR TITLE
chore: Add first unit tests

### PR DESCRIPTION
**Pull request by salisdemos**
for
- changeUrlForPager
- hasOwnPop
- hasPermission

### CODE REVIEW

File: tests/utils/hasOwnPop.spec.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import hasOwnProp from '../../utils/hasOwnProp'
```
Comment (gruenbergerdemos) [inline tests/utils/hasOwnPop.spec.js]: There is a typo in the test file name.

File: tests/utils/hasPermission.spec.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { hasPermission, hasAnyPermissions, hasAllPermissions } from '../../utils/hasPermission'
+
+window.dplan = {
+  permissions: {
+    myPermission: true,
+    notMyPermission: false,
+    someOtherPremission: true,
```
Comment (gruenbergerdemos) [inline tests/utils/hasPermission.spec.js]: Typo
```suggestion
    someOtherPermission: true,
```

File: tests/utils/hasPermission.spec.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  }
+}
+
+describe.each([
+  { permission: 'myPermission', result: true },
+  { permission: undefined, result: false },
+  { permission: null, result: false },
+  { permission: 'notIntheList', result: false },
+  { permission: ['myPermission'], result: true },
+  { permission: ['myPermission', 'someOtherPremission'], result: false },
```
Comment (gruenbergerdemos) [inline tests/utils/hasPermission.spec.js]:
```suggestion
  { permission: ['myPermission', 'someOtherPermission'], result: false },
```

File: tests/utils/hasPermission.spec.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  { permission: undefined, result: 'error' },
+])('hasPermission - check any access rights', ({ permission, result}) => {
+  test(`returns 'Error' for testcase with '${permission}'`, () => {
+    expect(() => hasAnyPermissions(permission)).toThrowError('Typeof "permissions" is not an Array')
+  })
+})
+
+describe.each([
+  { permission: ['myPermission', 'notHere', 'andNothingHere'], result: false },
+  { permission: ['myPermission', 'andIcanAccessEverything', 'someOtherPremission'], result: true },
```
Comment (gruenbergerdemos) [inline tests/utils/hasPermission.spec.js]:
```suggestion
  { permission: ['myPermission', 'andIcanAccessEverything', 'someOtherPermission'], result: true },
```

File: tests/utils/hasPermission.spec.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  { permission: ['myPermission', 'someOtherPremission'], result: false },
+])('hasPermission - check access rights for one permission', ({ permission, result}) => {
+  test(`returns ${result} for testcase with prop ${permission}`, () => {
+    expect(hasPermission(permission)).toBe(result)
+  })
+})
+
+describe.each([
+  { permission: ['myPermission', 'notHere', 'andNothingHere'], result: true },
+  { permission: ['myPermission', 'andIcanAccessEverything', 'someOtherPremission'], result: true },
```
Comment (gruenbergerdemos) [inline tests/utils/hasPermission.spec.js]:
```suggestion
  { permission: ['myPermission', 'andIcanAccessEverything', 'someOtherPermission'], result: true },
```

File: tests/utils/hasOwnPop.spec.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+])('hasOwnProp - a util to shorten Object.prototype.hasOwnPropery', ({ obj, prop, result, warn }) => {
+  const warnings = []
+
+  beforeEach(() => {
+    console.warn = msg => warnings.push(msg)
+  })
+
+  afterEach(() => {
+    console.warn = OGwarn
+  })
```
Comment (gruenbergerdemos) [inline tests/utils/hasOwnPop.spec.js]: I don't quite understand this block. Why do we need `OGwarn`?

File: tests/utils/changeUrlForPager.test.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        href: url
+      }
+    })
+
+    const updatedUrl = changeUrlforPager({ count: 3, current_page: 3 })
+    expect(updatedUrl[0]).toBe('http://dummy.com')
+    expect(updatedUrl[1]).toBe('r_limit=3&page=3')
+  })
+
+  test('without previous limit', () => {
```
Comment (gruenbergerdemos) [inline tests/utils/changeUrlForPager.test.js]: Did you mean `with previous limit`?

File: tests/utils/changeUrlForPager.test.js [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+        href: url
+      }
+    })
+
+    const updatedUrl = changeUrlforPager({ count: 3, current_page: 3 })
+    expect(updatedUrl[0]).toBe('http://dummy.com')
+    expect(updatedUrl[1]).toBe('r_limit=3&page=3')
+  })
+
+  test('without previous page', () => {
```
Comment (gruenbergerdemos) [inline tests/utils/changeUrlForPager.test.js]: Did you mean `with previous page`?

File: tests/utils/changeUrlForPager.test.js [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
```
Comment (gruenbergerdemos) [inline tests/utils/changeUrlForPager.test.js]: Did you mean `without previous page and limit`?

**Review by gruenbergerdemos — CHANGES_REQUESTED** [review]
Just some nitpicks. 
What is the difference between test.js and spec.js?

File: tests/utils/hasOwnPop.spec.js (hunk 1 above)
Comment (salisdemos) [reply tests/utils/hasOwnPop.spec.js]: 5616f7bd30909b91d72058f49a2ee555da4e8c6e

File: tests/utils/hasPermission.spec.js (hunk 2 above)
Comment (salisdemos) [reply tests/utils/hasPermission.spec.js]: 5616f7bd30909b91d72058f49a2ee555da4e8c6e

File: tests/utils/hasPermission.spec.js (hunk 3 above)
Comment (salisdemos) [reply tests/utils/hasPermission.spec.js]: 5616f7bd30909b91d72058f49a2ee555da4e8c6e

File: tests/utils/hasPermission.spec.js (hunk 5 above)
Comment (salisdemos) [reply tests/utils/hasPermission.spec.js]: 5616f7bd30909b91d72058f49a2ee555da4e8c6e

File: tests/utils/hasPermission.spec.js (hunk 4 above)
Comment (salisdemos) [reply tests/utils/hasPermission.spec.js]: 5616f7bd30909b91d72058f49a2ee555da4e8c6e

File: tests/utils/hasOwnPop.spec.js (hunk 6 above)
Comment (salisdemos) [reply tests/utils/hasOwnPop.spec.js]: Can't say either. This file is an old test. I just copied it

File: tests/utils/changeUrlForPager.test.js (hunk 7 above)
Comment (salisdemos) [reply tests/utils/changeUrlForPager.test.js]: I changed the naming entierly. Pleas have a look if it got better
8ccdc7eedb8c5b8064a40b1a5736c91cf45a3f86

File: tests/utils/changeUrlForPager.test.js (hunk 8 above)
Comment (salisdemos) [reply tests/utils/changeUrlForPager.test.js]: see above

File: tests/utils/changeUrlForPager.test.js (hunk 9 above)
Comment (salisdemos) [reply tests/utils/changeUrlForPager.test.js]: see above

File: tests/utils/changeUrlForPager.test.js [hunk 10]
```diff
@@ -0,0 +1,79 @@
+import changeUrlforPager from '../../utils/changeUrlforPager'
+
+describe('changeUrlforPager', () => {
+  beforeAll(() =>  {
+    global.window = Object.create(window);
+  })
+
+  it('should set givven Values if there are none already set', () => {
```
Comment (gruenbergerdemos) [inline tests/utils/changeUrlForPager.test.js]: Typo
```suggestion
  it('should set given values if there are none already set', () => {
```

File: tests/utils/changeUrlForPager.test.js [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+    const url = changeUrlforPager({ count: 2, current_page: 1 })
+
+    expect(url[1]).toBe("r_limit=2&page=1")
+  })
+
+  it('should overwrite givven Values if there are values already set', () => {
```
Comment (gruenbergerdemos) [inline tests/utils/changeUrlForPager.test.js]:
```suggestion
  it('should overwrite given values if there are values already set', () => {
```

File: tests/utils/changeUrlForPager.test.js [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+        href: url
+      }
+    })
+
+    const updatedUrl = changeUrlforPager({ count: 3, current_page: 3 })
+    expect(updatedUrl[0]).toBe('http://dummy.com')
+    expect(updatedUrl[1]).toBe('r_limit=3&page=3')
+  })
+
+  it('should overwrite givven limit if that is the only param', () => {
```
Comment (gruenbergerdemos) [inline tests/utils/changeUrlForPager.test.js]:
```suggestion
  it('should overwrite given limit if that is the only param', () => {
```

File: tests/utils/changeUrlForPager.test.js [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+        href: url
+      }
+    })
+
+    const updatedUrl = changeUrlforPager({ count: 3, current_page: 3 })
+    expect(updatedUrl[0]).toBe('http://dummy.com')
+    expect(updatedUrl[1]).toBe('r_limit=3&page=3')
+  })
+
+  it('should overwrite givven pge if that is the only param', () => {
```
Comment (gruenbergerdemos) [inline tests/utils/changeUrlForPager.test.js]:
```suggestion
  it('should overwrite given page if that is the only param', () => {
```

File: tests/utils/changeUrlForPager.test.js [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+        href: url
+      }
+    })
+
+    const updatedUrl = changeUrlforPager({ count: 3, current_page: 3 })
+    expect(updatedUrl[0]).toBe('http://dummy.com')
+    expect(updatedUrl[1]).toBe('r_limit=3&page=3')
+  })
+
+  it('should work even if there are other params otr the order ist mixed', () => {
```
Comment (gruenbergerdemos) [inline tests/utils/changeUrlForPager.test.js]:
```suggestion
  it('should work even if there are other params or the order is mixed', () => {
```

**Review by gruenbergerdemos — CHANGES_REQUESTED** [review]
Just some more typos, wording sounds good!

File: tests/utils/changeUrlForPager.test.js (hunk 14 above)
Comment (salisdemos) [reply tests/utils/changeUrlForPager.test.js]: where do all the t come from? :see_no_evil:

**Review by gruenbergerdemos — APPROVED** [review]
lgtm

**Pull request merged**